hlapipeline/alignimages.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python
+
+"""This script is a modernized replacement of tweakreg.
+
+"""
+
+import argparse
+from astropy.io import fits
+from astropy.table import Table
+import pdb
+from stwcs.wcsutil import HSTWCS
+import sys
+from utils import astrometric_utils as amutils
+
+# Module-level dictionary contains instrument/detector-specific parameters used later on in the script.
+detector_specific_params = {"acs":
+                                {"hrc":
+                                     {"fwhmpsf":0.073}, # TODO: Verify value
+                                 "sbc":
+                                     {"fwhmpsf":0.065}, # TODO: Verify value
+                                 "wfc":
+                                     {"fwhmpsf":0.076}}, # TODO: Verify value
+                            "wfc3":
+                                {"ir":{
+                                    "fwhmpsf":0.14}, # TODO: nothing. pipeline value in agreement with instrument handbook value.
+                                 "uvis":{
+                                     "fwhmpsf":0.076}}} # TODO: nothing. pipeline value in agreement with instrument handbook value.
+#-----------------------------------------------------------------------------------------------------------------------
+def main(input_list):
+    """Main calling function.
+    
+    Parameters
+    ----------
+    input_list: list
+        List of one or more input image(s) and/or association file(s) to align.
+
+    Returns
+    -------
+    Nothing for now.
+
+    """
+
+    # 1: Interpret input data and optional parameters
+
+    # 2: Apply filter to input observations to insure that they meet minimum criteria for being able to be aligned
+    imgList = input_list
+    # 3: Build WCS for full set of input observations
+    refwcs = amutils.build_reference_wcs(imgList)
+
+    # 4: Retrieve list of astrometric sources from database
+
+    # 5: Extract catalog of observable sources from each input image
+    extracted_sources = generate_source_catalogs(imgList,refwcs)
+
+    # 6: Cross-match source catalog with astrometric reference source catalog
+
+    # 7: Perform fit between source catalog and reference catalog
+#-----------------------------------------------------------------------------------------------------------------------
+def generate_source_catalogs(imgList,refwcs,**pars):
+    """Generates a dictionary of source catalogs keyed by image name.
+
+    Parameters
+    ----------
+    imgList: list
```

Comment:
One of the frustrating aspects of this format is how picky it is regarding the use of spaces.  Each colon used in the docstring for the parameter definition must be BOTH preceded and followed by a space in order for the formatting to be correct in the final HTML/PDF.